node/src/components/storage.rs

```diff
@@ -1612,7 +1753,7 @@ impl Storage {
     ) -> Option<SharedObject<Vec<u8>>> {
         if self.enable_mem_deduplication {
             // On a cache hit, return directly from the cache.
-            if let Some(serialized) = self.deploy_cache.get(&deploy_hash) {
+            if let Some(serialized) = self.deploy_cache.read().unwrap().get(&deploy_hash) {
```

Comment:
Please avoid panicking macros.

Reply:
Sadly, this is outside of well-maintained code paths at the moment, so a little out of scope (the functionality should be moved out of the routing code first, I do not want to return an error here because there will be more error handling code stuck in `participating.rs`).